opentreemap/importer/util.py

```diff
@@ -45,5 +45,9 @@ def utf8_file_to_csv_dictreader(f):
     # CSV standard "" to escape a double quote. Excel and LibreOffice
     # use this escape by default when saving as CSV.
     dialect.doublequote = True
-    return csv.DictReader(_as_utf8(f),
+
+    # the binary filestream (f) must be repackaged to the csv spec,
+    # that is an iterator of unicode strings
+    csv_body = StringIO(f.read().decode())
```

Comment:
I was concerned that we were now reading the entire CSV into memory when we were previously wrapping it with a reader that could iterate over it line by line. However, when I traced the code path backward from `utf8_file_to_csv_dictreader` I found that `process_csv` is already reading the entire file.

https://github.com/OpenTreeMap/otm-core/blob/053e5c17322719a80ee93c190c05df285890bc58/opentreemap/importer/views.py#L712

This may mean that we are going to have 2 copies of the entire file in memory while it is being sliced into individual database rows. That is probably ok.

Reply:
I considered the memory implications and it slipped my mind to vocalize that in the PR write-up, thanks for bringing it up.

Ideally, the CSV data would be read in as a stream in one encoding (like unicode `str` with StringIO) and have that be OK all the way through but that is not the case. In earlier CSV processing steps, a bytestream is necessary but then a string stream is needed here. Surely there is a more robust way to handle this than holding up to two copies of the data in memory, but I think that would involve much more time spent in the code base. I have no idea how big user-uploaded CSVs are IRL, but I am glad for the sanity check from your comment that this may not be too big a burden for this code pattern.

I will update the commit message with detail of this consideration of memory.